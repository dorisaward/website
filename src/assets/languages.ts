import { I18n } from 'aws-amplify/utils'

export const setLanguage = I18n.setLanguage

export const getText = (key: Dictionary) => I18n.get(key)

export enum languages {
    en = 'ENGLISH',
    th = 'THAI',
  }

type Dictionary = 'sorry'

type Dictionaries = Record<languages, Record<Dictionary, string>>

export const dictionary: Dictionaries = {
    [languages.en]: {
        'sorry': 'Sorry'
    },
    [languages.th]: {
        'sorry': 'ขอโทษ'
    },
}

export const setUpLanguages = () => {
    I18n.putVocabularies(dictionary)
    setLanguage(languages.en)
}
