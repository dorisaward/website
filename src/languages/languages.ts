export enum languages {
    en = 'ENGLISH',
    th = 'THAI',
}
