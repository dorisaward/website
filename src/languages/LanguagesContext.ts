import { createContext } from 'react'
import { languages } from './languages.ts'
import { LangContext } from './types/LanguagesContext.ts'

export const LanguagesContext = createContext<LangContext>({
    language: languages.en,
    getText: () => '',
    handleLangPress: () => {},
})