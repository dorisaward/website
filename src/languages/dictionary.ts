import { languages } from './languages.ts'
import { Dictionaries } from './types/Dictionaries.ts'

export const dictionary: Dictionaries = {
    [languages.en]: {
        'sorry': 'Sorry',
        'error': 'Nothing to see here',
        'home': 'Home',
        'cv': 'Curriculum Vitae',
    },
    [languages.th]: {
        'sorry': 'ขอโทษ',
        'error': 'ไม่มีที่นี่',
        'home': 'บ้าน',
        'cv': 'ประวัติย่อ',
    },
}