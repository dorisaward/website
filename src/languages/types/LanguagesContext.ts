import { languages } from '../languages.ts'
import { Dictionary } from './Dictionary.ts'

export type LangContext = {
    language: languages
    getText: (key: Dictionary) => string
    handleLangPress: () => void
}