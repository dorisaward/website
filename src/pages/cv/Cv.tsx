import { use } from 'react'
import { LanguagesContext } from '../../languages/LanguagesContext.ts'
import { languages } from '../../languages/languages.ts'

export const Cv = () => {
    const { language } = use(LanguagesContext)
    return (
        <>
            {language === languages.th && <p>Unfortunately, Thai language is not available</p>}
            <p>Curriculum Vitae</p>
        </>
    )
}