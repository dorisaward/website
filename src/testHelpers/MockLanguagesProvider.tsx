import { MockElementPropsType } from './MockElementPropsType.ts'
import { LanguagesContext } from '../languages/LanguagesContext.ts'
import { LangContext } from '../languages/types/LanguagesContext.ts'

export const MockLanguagesProvider = ({ children, value }: MockElementPropsType<{ value: LangContext }>) => (
    <LanguagesContext value={value}>
        {children}
    </LanguagesContext>
)
