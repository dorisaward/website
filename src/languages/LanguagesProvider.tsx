import { FC, useState, useEffect, ReactNode, useMemo } from 'react'
import { I18n } from 'aws-amplify/utils'
import { Dictionary } from './types/Dictionary.ts'
import { dictionary } from './dictionary.ts'
import { languages } from './languages.ts'
import { LanguagesContext } from './LanguagesContext.ts'
import { LangContext } from './types/LanguagesContext.ts'

const setLanguage = I18n.setLanguage

const setUpLanguages = () => {
    I18n.putVocabularies(dictionary)
    setLanguage(languages.en)
}

export const LanguagesProvider: FC<{ children: ReactNode }> = ({ children }) => {
    const [selectedLang, setSelectedLang] = useState<languages>(languages.en)
    useEffect(setUpLanguages, [])

    const value = useMemo<LangContext>(() => ({
        language: selectedLang,
        getText: (key: Dictionary) => I18n.get(key),
        handleLangPress: () => setSelectedLang(prevLang => {
            const newLang = prevLang === languages.en ? languages.th : languages.en
            setLanguage(newLang)
            return newLang
        })
    }), [ selectedLang ])

    return (
        <LanguagesContext value={value}>
            {children}
        </LanguagesContext>
    )
}