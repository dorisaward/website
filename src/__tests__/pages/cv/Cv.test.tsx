import { render } from '@testing-library/react'
import { Cv } from '../../../pages/cv/Cv.tsx'
import { LangContext } from '../../../languages/types/LanguagesContext.ts'
import { MockLanguagesProvider } from '../../../testHelpers/MockLanguagesProvider.tsx'
import { languages } from '../../../languages/languages.ts'

jest.mock('../../../assets/cv.json', () => ({
    'basics': {
        'name': 'basic name',
        'label': 'label',
        'email': 'email',
        'url': 'url',
        'phone': 'phone',
        'location': {
            'city': 'city',
            'countryCode': 'country'
        },
        'summary': 'basic summary',
        'profiles': [{
            'network': 'network',
            'username': 'username',
            'url': 'profile url'
        }]
    },
    'work': [{
        'name': 'work name',
        'position': 'position',
        'url': 'work url',
        'startDate': 'work start',
        'summary': 'work summary',
    }],
    'education': [{
        'institution': 'institution',
        'url': 'edu url',
        'area': 'area',
        'studyType': 'study',
        'startDate': 'edu start',
        'endDate': 'edu end'
    }],
    'skills': [{
        'name': 'skill name',
        'level': 'level',
        'keywords': [
            'keyword'
        ]
    }],
    'languages': [{
        'language': 'lang',
        'fluency': 'fluency'
    }]
}))
jest.mock('../../../pages/cv/logos/index.ts', () => ({
    images: [
        jest.mock('../../../pages/cv/logos/react.png'),
        jest.mock('../../../pages/cv/logos/react.png')
    ]
}))

const thaiLangError = 'Unfortunately, Thai language is not available'

describe('Cv', () => {
    it('renders, given language english', () => {
        // Given
        const value: LangContext = {
            getText: jest.fn(),
            handleLangPress: jest.fn(),
            language: languages.en
        }

        // When
        const { baseElement, queryByText, getAllByRole } = render(
            <MockLanguagesProvider value={value}><Cv/></MockLanguagesProvider>
        )

        // Then
        expect(queryByText(thaiLangError)).toBeFalsy()
        expect(getAllByRole('img')).toHaveLength(2)
        expect(baseElement).toMatchSnapshot()
    })

    it('renders, given language thai', () => {
        // Given
        const value: LangContext = {
            getText: jest.fn(),
            handleLangPress: jest.fn(),
            language: languages.th
        }

        // When
        const { getByText, baseElement, getAllByRole } = render(
            <MockLanguagesProvider value={value}><Cv/></MockLanguagesProvider>
        )

        // Then
        expect(getByText(thaiLangError)).toBeTruthy()
        expect(getAllByRole('img')).toHaveLength(2)
        expect(baseElement).toMatchSnapshot()
    })
})
