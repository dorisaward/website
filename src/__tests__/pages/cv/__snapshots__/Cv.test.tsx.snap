// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cv renders, given language english 1`] = `
<body>
  <div>
    <div
      class="sc-blHHSb jMgSmr"
    >
      <h1>
        Curriculum Vitae
      </h1>
      <div
        class="sc-gtLWhw iREGgd"
      >
        <h2>
          label
        </h2>
        <p>
          basic summary
        </p>
      </div>
      <div
        class="sc-gtLWhw iREGgd"
      >
        <h2>
          Work
        </h2>
        <p>
          work name - work url
work start - Current
position
work summary
        </p>
      </div>
      <div
        class="sc-gtLWhw iREGgd"
      >
        <h2>
          Contact
        </h2>
        <p>
          email
url
phone
city, country
References available on request
        </p>
      </div>
      <div
        class="sc-gtLWhw iREGgd"
      >
        <h2>
          Education
        </h2>
        <p>
          edu start - edu end
study - area
institution - edu url
        </p>
      </div>
      <div
        class="sc-egkSDF dzeDsK"
      >
        <img
          class="sc-fAUdSK ldTxjN"
          src="[object Object]"
        />
        <img
          class="sc-fAUdSK ldTxjN"
          src="[object Object]"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Cv renders, given language thai 1`] = `
<body>
  <div>
    <div
      class="sc-blHHSb jMgSmr"
    >
      <p>
        Unfortunately, Thai language is not available
      </p>
      <h1>
        Curriculum Vitae
      </h1>
      <div
        class="sc-gtLWhw iREGgd"
      >
        <h2>
          label
        </h2>
        <p>
          basic summary
        </p>
      </div>
      <div
        class="sc-gtLWhw iREGgd"
      >
        <h2>
          Work
        </h2>
        <p>
          work name - work url
work start - Current
position
work summary
        </p>
      </div>
      <div
        class="sc-gtLWhw iREGgd"
      >
        <h2>
          Contact
        </h2>
        <p>
          email
url
phone
city, country
References available on request
        </p>
      </div>
      <div
        class="sc-gtLWhw iREGgd"
      >
        <h2>
          Education
        </h2>
        <p>
          edu start - edu end
study - area
institution - edu url
        </p>
      </div>
      <div
        class="sc-egkSDF dzeDsK"
      >
        <img
          class="sc-fAUdSK ldTxjN"
          src="[object Object]"
        />
        <img
          class="sc-fAUdSK ldTxjN"
          src="[object Object]"
        />
      </div>
    </div>
  </div>
</body>
`;
